lib/std/os/uefi/protocols/device_path_protocol.zig

```diff
@@ -292,130 +402,288 @@ pub const MessagingDevicePath = union(Subtype) {
+        std.debug.assert(1 == @offsetOf(F1394DevicePath, "subtype"));
+        std.debug.assert(2 == @offsetOf(F1394DevicePath, "length"));
+        std.debug.assert(4 == @offsetOf(F1394DevicePath, "reserved"));
+        std.debug.assert(8 == @offsetOf(F1394DevicePath, "guid"));
+    }
+
+    pub const UsbDevicePath = extern struct {
         type: DevicePathType,
         subtype: Subtype,
-        length: u16,
+        length: u16 align(1),
         parent_port_number: u8,
         interface_number: u8,
     };
 
-    pub const SataDevicePath = packed struct {
+    comptime {
+        std.debug.assert(6 == @sizeOf(UsbDevicePath));
+        std.debug.assert(1 == @alignOf(UsbDevicePath));
+
+        std.debug.assert(0 == @offsetOf(UsbDevicePath, "type"));
+        std.debug.assert(1 == @offsetOf(UsbDevicePath, "subtype"));
+        std.debug.assert(2 == @offsetOf(UsbDevicePath, "length"));
+        std.debug.assert(4 == @offsetOf(UsbDevicePath, "parent_port_number"));
+        std.debug.assert(5 == @offsetOf(UsbDevicePath, "interface_number"));
+    }
+
+    pub const SataDevicePath = extern struct {
         type: DevicePathType,
         subtype: Subtype,
-        length: u16,
-        hba_port_number: u16,
-        port_multiplier_port_number: u16,
-        logical_unit_number: u16,
+        length: u16 align(1),
+        hba_port_number: u16 align(1),
+        port_multiplier_port_number: u16 align(1),
+        logical_unit_number: u16 align(1),
     };
 
-    pub const UsbWwidDevicePath = packed struct {
+    comptime {
+        std.debug.assert(10 == @sizeOf(SataDevicePath));
+        std.debug.assert(1 == @alignOf(SataDevicePath));
+
+        std.debug.assert(0 == @offsetOf(SataDevicePath, "type"));
+        std.debug.assert(1 == @offsetOf(SataDevicePath, "subtype"));
+        std.debug.assert(2 == @offsetOf(SataDevicePath, "length"));
+        std.debug.assert(4 == @offsetOf(SataDevicePath, "hba_port_number"));
+        std.debug.assert(6 == @offsetOf(SataDevicePath, "port_multiplier_port_number"));
+        std.debug.assert(8 == @offsetOf(SataDevicePath, "logical_unit_number"));
+    }
+
+    pub const UsbWwidDevicePath = extern struct {
         type: DevicePathType,
         subtype: Subtype,
-        length: u16,
-        interface_number: u16,
-        device_vendor_id: u16,
-        device_product_id: u16,
+        length: u16 align(1),
+        interface_number: u16 align(1),
+        device_vendor_id: u16 align(1),
+        device_product_id: u16 align(1),
 
         pub fn serial_number(self: *const UsbWwidDevicePath) []const u16 {
             var serial_len = (self.length - @sizeOf(UsbWwidDevicePath)) / @sizeOf(u16);
-            return @ptrCast([*]u16, @ptrCast([*]u8, self) + @sizeOf(UsbWwidDevicePath))[0..serial_len];
+            return @ptrCast(
+                [*]const u16,
+                @alignCast(2, @ptrCast([*]const u8, self) + @sizeOf(UsbWwidDevicePath)),
+            )[0..serial_len];
         }
     };
```

Comment:
Same question about alignment here (and in a few other places).

Reply:
DevicePaths are not aligned in memory, but strings must be

Reply:
AlignCast doesn't change the alignment of this though. You should have an under-aligned pointer instead, surely.

Reply:
Given I can't find anything about how this string is aligned and that it directly follows this struct, I think its correct to underalign this